### PR TITLE
The cluster coordinator support watch mechanism for notifying `RemoteClientManager` and `ServerStatusService`.

### LINKED ISSUE #10076: [Feature] Support listeners for the `RemoteClients` change.
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/skywalking/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

    In the cluster mode, OAP instances send metrics to each other according to the `remote-hash code` to do L2 aggregation and alarm calculations for balancing​. Some of the calculations leverage the previously received metrics in the memory cache and incoming metrics. So when the OAP cluster scales,  the metrics might lose their cache and impact the calculations.
   To avoid this situation, I think we should support listeners for the `RemoteClients` change, then the stream workers can be notified and reload previous metrics from storage for calculation.

### Use case

1. Provide listeners' interface for notification and a registration mechanism.
2. When the `RemoteClients` changes, notice all listeners.

### Related issues

https://github.com/apache/skywalking/issues/10051

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by wankai123**
- [ ] If this is non-trivial feature, paste the links/URLs to the design doc.
- [ ] Update the documentation to include this new feature.
- [X] Tests(including UT, IT, E2E) are added to verify the new feature.
- [ ] If it's UI related, attach the screenshots below.

- [ ] If this pull request closes/resolves/fixes an existing issue, replace the issue number. Closes #10076.
- [X] Update the [`CHANGES` log](https://github.com/apache/skywalking/blob/master/docs/en/changes/changes.md).

also relate to #9814 
I reserve the query timer mechanism and extended the interval, just in case the notification from the third-party client fails due to network or other issues.

### CODE REVIEW

File: oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/CoreModuleProvider.java [hunk 1]
```diff
@@ -355,18 +355,20 @@ public void start() throws ModuleStartException {
 
         Address gRPCServerInstanceAddress = new Address(moduleConfig.getGRPCHost(), moduleConfig.getGRPCPort(), true);
         TelemetryRelatedContext.INSTANCE.setId(gRPCServerInstanceAddress.toString());
+        ClusterCoordinator coordinator = this.getManager()
+                                             .find(ClusterModule.NAME)
+                                             .provider()
+                                             .getService(ClusterCoordinator.class);
+        coordinator.startCoordinator();
+        coordinator.registerWatcher(remoteClientManager);
```
Comment (wu-sheng) [inline oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/CoreModuleProvider.java:363]: Should we do register first before start coodinator?

File: oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/cluster/ClusterModule.java [hunk 2]
```diff
@@ -32,7 +32,9 @@ public ClusterModule() {
     public Class[] services() {
         return new Class[] {
             ClusterRegister.class,
-            ClusterNodesQuery.class
+            ClusterNodesQuery.class,
+            ClusterWatcherRegister.class,
+            ClusterCoordinator.class
```
Comment (wu-sheng) [inline oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/cluster/ClusterModule.java:37]: As `ClusterCoordinator` inherits `ClusterWatcherRegister`, do we need to expose both?

File: oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/CoreModuleProvider.java (hunk 1 above)
Comment (wankai123) [reply oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/CoreModuleProvider.java:363]: Not necessary, but do register first more clearly. I'll fix it

File: oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/remote/client/RemoteClientManager.java [hunk 3]
```diff
@@ -93,14 +94,26 @@ public RemoteClientManager(final ModuleDefineHolder moduleDefineHolder, final in
...
+            this.clusterNodesQuery = moduleDefineHolder.find(ClusterModule.NAME)
+                                                       .provider()
+                                                       .getService(ClusterNodesQuery.class);
+        }
+
+        this.refresh(clusterNodesQuery.queryRemoteNodes());
     }
 
     /**
      * Query OAP server list from the cluster module and create a new connection for the new node. Make the OAP server
```
Comment (wu-sheng) [inline oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/remote/client/RemoteClientManager.java:113]: Comments should be updated accordingly.

File: oap-server/server-cluster-plugin/cluster-zookeeper-plugin/src/main/java/org/apache/skywalking/oap/server/cluster/plugin/zookeeper/ZookeeperCoordinator.java [hunk 4]
```diff
@@ -128,4 +134,30 @@ private void initHealthChecker() {
             healthChecker = metricCreator.createHealthCheckerGauge("cluster_zookeeper", MetricsTag.EMPTY_KEY, MetricsTag.EMPTY_VALUE);
         }
     }
+
+    @Override
+    protected void start() {
+        initHealthChecker();
+        serviceCache.addListener(new ZookeeperEventListener());
```
Comment (kezhenxu94) [inline oap-server/server-cluster-plugin/cluster-zookeeper-plugin/src/main/java/org/apache/skywalking/oap/server/cluster/plugin/zookeeper/ZookeeperCoordinator.java:141]: Maybe move the line 64 (`this.serviceCache.start();`) here and move this line to the constructor?

File: oap-server/server-cluster-plugin/cluster-zookeeper-plugin/src/main/java/org/apache/skywalking/oap/server/cluster/plugin/zookeeper/ZookeeperCoordinator.java [hunk 5]
```diff
@@ -115,6 +117,10 @@ public List<RemoteInstance> queryRemoteNodes() {
             this.healthChecker.unHealth(e);
             throw new ServiceQueryException(e.getMessage());
         }
+
+        if (log.isDebugEnabled()) {
+            remoteInstances.forEach(instance -> log.debug("Zookeeper cluster instance: {}", instance.toString()));
```
Comment (kezhenxu94) [inline oap-server/server-cluster-plugin/cluster-zookeeper-plugin/src/main/java/org/apache/skywalking/oap/server/cluster/plugin/zookeeper/ZookeeperCoordinator.java:122]:
```suggestion
            remoteInstances.forEach(instance -> log.debug("Zookeeper cluster instance: {}", instance));
```

**Review by kezhenxu94 — COMMENTED** [review]
Some nits, others LGTM

**Review by kezhenxu94 — APPROVED** [review]

**Review by wu-sheng — APPROVED** [review]
lgtm. In the future, we may have to decide which are recommended ways to use coordinator mechanism. Query list and watch exist at the same time, could cause confusion.

**Pull request merged**